skimage/measure/_regionprops.py

```diff
@@ -62,9 +64,56 @@
     'WeightedCentralMoments': 'weighted_moments_central',
     'WeightedCentroid': 'weighted_centroid',
     'WeightedHuMoments': 'weighted_moments_hu',
+    'WeightedLocalCentroid': 'weighted_local_centroid',
     'WeightedMoments': 'weighted_moments',
     'WeightedNormalizedMoments': 'weighted_moments_normalized'
 }
+object_columns = {'image', 'coords', 'convex_image', 'filled_image',
+               'intensity_image'}
+
+COL_DTYPES = {
```

Comment:
@jni Can this be accomplished using annotations?

Reply:
@stefanv a very interesting idea! Let me think about it! It might be a bit tricky because of how convoluted regionprops is (using `@property` and `@_cached_property`), but I think it could work! I think the hardest thing will be dealing with `Array[float]`. Dunno whether that's a thing.

Reply:
It's a thing! https://github.com/machinalis/mypy-data/blob/master/numpy-mypy/numpy/__init__.pyi

Reply:
@stefanv sure, but then we need to depend on that file... Should we vendor it? But I must admit it would be quite appealing. I think we could annotate the `_RegionProps` properties as:

```python
def inertia_tensor(self) -> ndarray[float]:
```

And I think they could then be accessed with `_RegionProps.__annotations__`, which would be equivalent to this dict.

btw, should we rename `_RegionProps` to `RegionProps`? It's a private class in that users shouldn't directly instantiate it, but it's decidedly public in that they interact with the class extensively.

Reply:
I've been trying to use annotations, but they haven't really been fun to use with numpy.

Reply:
@leGIT-bot as an aside, @stefanv told me in a separate channel (sorry! so much out-of-context stuff going on in this PR...) that he supports renaming `_RegionProperties` to `RegionProperties`. To do this, rename the class, replacing all internal usages with the new name (VSCode and PyCharm both have facilities to make this trivial), and then add an alias `_RegionProperties = RegionProperties` immediately below the class definition, with a comment that this is for backwards compatibility.

With regards to using proper type annotations: @stefanv, I think that this should come in a later PR. It's a complicated process due to the vendoring of array definitions, so I think it would delay this PR for an unnecessarily long time just for an implementation detail.

Having said that, @leGIT-bot, I suggest replacing `'int'` with `int`, `'float'` with `float`, and `'object'` with `object`, in this dictionary.

Reply:
and `'slice'` with `slice`. Which I believe should be added to the `objects` columns set.